MosipNexus/crawler/web_crawler.py

```
@@ -0,0 +1,225 @@
+def get_all_page_urls(sitemap_url: str, visited: Optional[set] = None) -> list[str]:
+    """Follow sitemap indexes recursively; return only actual page URLs."""
+    if visited is None:
+        visited = set()
+    if sitemap_url in visited:
+        return []
+    visited.add(sitemap_url)
+
+    try:
+        res = requests.get(sitemap_url, headers=HTTP_HEADERS, timeout=30)
+        res.raise_for_status()
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)